internal/services/compute/shared_schema.go

```diff
@@ -641,7 +641,7 @@ func isValidHotPatchSourceImageReference(referenceInput []interface{}, imageId s
 	offer := raw["offer"].(string)
 	sku := raw["sku"].(string)
 
-	if pub == "MicrosoftWindowsServer" && offer == "WindowsServer" && (sku == "2022-datacenter-azure-edition-core" || sku == "2022-datacenter-azure-edition-core-smalldisk" || sku == "2022-datacenter-azure-edition-hotpatch" || sku == "2022-datacenter-azure-edition-hotpatch-smalldisk") {
+	if pub == "MicrosoftWindowsServer" && offer == "WindowsServer" && (sku == "2022-datacenter-azure-edition-core" || sku == "2022-datacenter-azure-edition-core-smalldisk" || sku == "2022-datacenter-azure-edition-hotpatch" || sku == "2022-datacenter-azure-edition-hotpatch-smalldisk" || sku == "2025-datacenter-azure-edition" || sku == "2025-datacenter-azure-edition-smalldisk" || sku == "2025-datacenter-azure-edition-core" || sku == "2025-datacenter-azure-edition-core-smalldisk") {
```

Comment:
This is verging on becoming unreadable and difficult to see at first glance which sku's support hotpatching, can we separate the sku comparisons out into a separate check, something like
```suggestion
	supportedSkus := []string{
	// list skus here
	}
	
	hotPatchingSupported := false
	
	for _, s := range supportedSkus {
		if sku == s {
			hotPatchingSupported = true
		}
	}
	
	if pub == "MicrosoftWindowsServer" && offer == "WindowsServer" && hotPatchingSupported {
```

Reply:
Updated. I found the [standard lib](https://pkg.go.dev/slices#Contains) already supports it so that I directly leverage it.